flamedisx/inference.py

```diff
@@ -360,54 +364,85 @@ def bestfit(self,
 
-        # Unfortunately we can only set the relative tolerance for the
-        # objective; we'd like to set the absolute one.
-        # Use the guess log likelihood to normalize;
-        if llr_tolerance is not None:
-            kwargs.setdefault('f_relative_tolerance',
-                              llr_tolerance/self.minus_ll(**guess)[0])
-
         # Create a vector of guesses for the optimizer
         # Store as temp attributes so we can access them also in objective
         self._varnames = [k for k in self.param_names if k not in fix]
         self._guess_vect = fd.np_to_tf(np.array([
             guess[k] for k in self._varnames]))
         self._fix = fix
-
-        # Minimize multipliers to the guess, rather than the guess itself
-        # This is a basic kind of standardization that helps make the gradient
-        # vector reasonable.
-        x_norm = tf.ones(len(self._varnames), dtype=fd.float_type())
-
+        
         if optimizer == tfp.optimizer.bfgs_minimize and use_hessian:
             # This optimizer can use the hessian information
             # Compute the inverse hessian at the guess
             inv_hess = self.inverse_hessian(guess, omit_grads=tuple(fix.keys()))
-            # We use scaled values in the optimizer so also scale the
-            # hessian. We need to multiply the hessian with the parameter
-            # values. This is the inverse hessian so we divide.
-            inv_hess /= tf.linalg.tensordot(
-                self._guess_vect, self._guess_vect, axes=0)
             # Explicitly symmetrize the matrix
             inv_hess = fd.symmetrize_matrix(inv_hess)
         else:
             inv_hess = None
 
         self._autograph_objective = autograph
-        res = optimizer(self.objective,
-                        x_norm,
-                        initial_inverse_hessian_estimate=inv_hess,
-                        **kwargs)
-        if get_lowlevel_result:
-            return res
-        if res.failed:
-            raise ValueError(f"Optimizer failure! Result: {res}")
-        res = res.position * self._guess_vect
-        res = {k: res[i].numpy() for i, k in enumerate(self._varnames)}
-        return {**res, **fix}
-
-    def objective(self, x_norm):
-        x = x_norm * self._guess_vect
+
+        if optimizer == Minuit.from_array_func:
+            x_guess =  np.array([
+            guess[k] for k in self._varnames])
+
+            fit = optimizer(self.objective_minuit,
+                           x_guess,
+                           grad=self.grad_minuit,
+                           errordef=0.5,
+                           name = self._varnames,
+                           **kwargs)
+            
+            fit.migrad(precision=precision)
+            fit_result = dict(fit.values)
+            fit_errors = dict()
+            for (k, v) in fit.errors.items():
+                fit_errors[k + '_error'] = v
+
+            return fit_result, fit_errors
```

Comment:
Will look into it!

Reply:
The tensorflow optimiser doesn't return errors... I think is best if we keep the different outputs.
This way the Minuit optimizer returns a tuple of dicts, one with parameter values (that can directly be used in the `summary` as `summary(bf[0])`) and the other with the parameter errors (with the Hesse errors if `use_hessian = True`).
The only difference in the output is that one uses `bf[0]` instead of `bf` with the results from Minuit

Reply:
Good point, perhaps we could add an empty dictionary in the output of the tensorflow minimizer? Just so we don't have to change any downstream code to switch between minimizers.

Reply:
Indeed! I am changing also the tutorial so that it looks at bf[0].

Reply:
done